expyfun/_input_controllers.py

```diff
@@ -489,3 +532,23 @@ def _retrieve_events(self, live_keys):
             if live_keys is None or key[0] in live_keys:
                 targets.append(key)
         return targets
+
+    def _retrieve_keyboard_releases(self, live_keys):
+        # add escape keys
+        if live_keys is not None:
+            live_keys = [str(x) for x in live_keys]  # accept ints
+            live_keys.extend(self.force_quit_keys)
+        # pump for events
+        self._dev.poll_for_response()
+        while self._dev.response_queue_size() > 0:
+            key = self._dev.get_next_response()
+            if not key['pressed']:
```

Comment:
I think that when you poll for events, *all* events get cleared from the `pyxid` buffer. So whenever polling for releases *or* presses, you'll need to append to our internal `self._keyboard_buffer_*` lists. This function will currently drop the presses, and the press function will drop the releases, right?

But yes, I'll be able to test this when I get to Seattle. The more annoying part will be modifying the TDT circuit. I'll probably have to do that, but it woudl be good if this were theoretically correct for Cedrus (and verified correct for Pyglet) by the time I'm there.